### PR TITLE
Implement process.env.NODE_ENV support

### PR DESCRIPTION
**Pull request by AleksandrSl**
https://cultofmartians.com/tasks/dual-publish-process-env.html

### CODE REVIEW

File: process-dir.js [hunk 1]
```diff
@@ -184,6 +196,41 @@ async function replacePackage (dir, file, files) {
   await writeFile(pkgFile, JSON.stringify(pkg, null, 2))
 }
 
+function hasNodeEnvCondition (source) {
+  return /process.env.NODE_ENV\s*[!=]==\s*["']production["']/.test(source)
+}
+
+async function replaceNodeEnvConditions (dir, file, source) {
+  source = source.toString()
+  let prodCondition = /process.env.NODE_ENV\s*===\s*["']production["']/g
```
Comment (AleksandrSl) [inline process-dir.js:205]: Maybe we should also respect `process.env.NODE_ENV === "development"`?

File: process-dir.js [hunk 2]
```diff
@@ -239,18 +286,34 @@ async function process (dir) {
...
   await Promise.all(
     sources.map(async ([file, source]) => {
       if (file.endsWith('index.browser.js')) {
-        await replaceToESM(dir, file, source)
+        let [, modifiedSource] = await replaceToESM(dir, file, source)
+        if (nodeEnvReplacementTargets.includes(file)) {
+          files.push(
+            ...(await replaceNodeEnvConditions(dir, file, modifiedSource))
+          )
+        }
```
Comment (AleksandrSl) [inline process-dir.js:301]: Should we remove `'index.browser.js'` after replacement?

File: process-dir.js [hunk 3]
```diff
@@ -146,7 +147,15 @@ async function replacePackage (dir, file, files) {
   pkg.module = 'index.js'
   pkg['react-native'] = 'index.js'
 
-  if (files.includes(file.replace(/\.js$/, '.browser.js'))) {
+  if (
+    nodeEnvReplacementTargets.includes(file) ||
```
Comment (ai) [inline process-dir.js:151]: Too long. Can we replaced it to `envTargets`?

File: process-dir.js (hunk 1 above)
Comment (ai) [reply process-dir.js:216]: Yeap, we need it too

File: process-dir.js (hunk 3 above)
Comment (AleksandrSl) [reply process-dir.js:151]: Sure. Does it increase package size or it's just personal preference?

Comment (ai) [reply process-dir.js:151]: Just code style consistancy

File: process-dir.js (hunk 1 above)
Comment (AleksandrSl) [reply process-dir.js:216]: Maybe someone uses not strict equality, i.e. `==` will add it too

Comment (AleksandrSl) [reply process-dir.js:216]: And maybe backticks should be considered?

File: process-dir.js (hunk 3 above)
Comment (AleksandrSl) [reply process-dir.js]: Shortened this and other names to be consistent

File: process-dir.js (hunk 1 above)
Comment (AleksandrSl) [reply process-dir.js]: Added support for "development", not strict equality and strings in backticks

File: process-dir.js (hunk 2 above)
Comment (ai) [reply process-dir.js:302]: Yeap, if `index.browser.js` had `process.env.NODE_ENV` we should have only `index.dev.js`/`index.prod.js`

**Comment by ai** [comment]
Released in 1.0

**Pull request merged**